components/resourceDetails/resourceSharing/agentAccess/agentAccessOptionsMenu/index.test.jsx

```diff
@@ -53,3 +64,108 @@ describe("AgentAccessOptionsMenu", () => {
     expect(removeButton).toBeDefined();
   });
 });
+
+describe("Consent Details Modal", () => {
+  it("renders a details modal when you click on the view details button", async () => {
+    const permission = {
+      webId,
+      alias: "Editors",
+      type: "agent",
+      vc: getSignedVc(),
+    };
+    const { asFragment, getByTestId, queryByText } = renderWithTheme(
+      <AgentAccessOptionsMenu
+        resourceIri={resourceIri}
+        permission={permission}
+        setLoading={jest.fn}
+        setLocalAccess={jest.fn}
+      />
+    );
+
+    const menuButton = getByTestId("menu-button");
+    userEvent.click(menuButton);
+    await waitFor(() => {
+      const viewDetailsButton = getByTestId(TESTCAFE_ID_VIEW_DETAILS_BUTTON);
+      expect(viewDetailsButton).toBeInTheDocument();
+      userEvent.click(viewDetailsButton);
+    });
+
+    await waitFor(() => {
+      expect(
+        getByTestId(TESTCAFE_ID_CONSENT_DETAILS_MODAL)
+      ).toBeInTheDocument();
+    });
+  });
+
+  it("closes the details modal when you click on the revoke button", async () => {
+    const permission = {
+      webId,
+      alias: "Editors",
+      type: "agent",
+      vc: getSignedVc(),
+    };
+    jest.spyOn(accesGrantFns, "revokeAccessGrant").mockResolvedValue(true);
+
+    const { getByTestId } = renderWithTheme(
+      <AgentAccessOptionsMenu
+        resourceIri={resourceIri}
+        permission={permission}
+        setLoading={jest.fn}
+        setLocalAccess={jest.fn}
+      />
+    );
+
+    const menuButton = getByTestId("menu-button");
+    userEvent.click(menuButton);
+
+    await waitFor(() => {
+      const viewDetailsButton = getByTestId(TESTCAFE_ID_VIEW_DETAILS_BUTTON);
+      userEvent.click(viewDetailsButton);
+    });
+    const revokeButton = getByTestId(TESTCAFE_ID_CONSENT_DETAILS_REVOKE_BUTTON);
+    userEvent.click(revokeButton);
+
+    userEvent.click(revokeButton);
+    await waitFor(() => {
+      const closedModal = screen.queryByTestId(
+        TESTCAFE_ID_CONSENT_DETAILS_MODAL
+      );
+      expect(closedModal).toBeNull();
+    });
+  });
+  it("closes the details modal when you click on the done button", async () => {
```

Comment:
nitpick: Looks like you missed a space here.

Reply:
fixed!